src/avs/AVSList.jsx

```diff
@@ -1,194 +1,311 @@
     if (currentPage - 1 >= 1) {
       setSearchParams({ page: (currentPage - 1).toString() });
-      fetchAVS(currentPage - 1);
     }
-  }, [searchParams, fetchAVS]);
+  }, [searchParams, setSearchParams]);
 
   const handlePageClick = useCallback(
     page => {
       setSearchParams({ page: page.toString() });
-      fetchAVS(page);
     },
-    [setSearchParams, fetchAVS]
+    [setSearchParams]
   );
 
+  const handleSearch = e => {
+    dispatch({ searchTerm: e.target.value });
+  };
+
+  const truncate = str => {
+    return str.length > 42 ? str.substring(0, 42) + '...' : str;
+  };
+
   useEffect(() => {
-    const page = searchParams.get('page');
-    if (!page) {
-      setSearchParams({ page: 1 }, { replace: true });
-      fetchAVS(1);
-    } else fetchAVS(searchParams.get('page'));
-  }, [searchParams]);
+    const page = searchParams.get('page') ?? 1;
+
+    const params = {};
+    params.page = state.searchTriggered ? 1 : page; // If user has searched something update the page number to 1
+    if (debouncedSearchTerm) params.search = debouncedSearchTerm;
+    if (state.sortDescriptor) {
+      params.sort =
+        state.sortDescriptor.direction === 'ascending'
+          ? state.sortDescriptor.column
+          : `-${state.sortDescriptor.column}`;
+    }
+
+    setSearchParams(params, { replace: true });
+    fetchAVS(params.page, params.search, params.sort);
+    dispatch({ searchTriggered: false });
+  }, [
+    debouncedSearchTerm,
+    dispatch,
+    fetchAVS,
+    searchParams,
+    setSearchParams,
+    state.searchTriggered,
+    state.sortDescriptor
+  ]);
+
+  useEffect(() => {
+    if (debouncedSearchTerm) dispatch({ searchTriggered: true });
+  }, [dispatch, debouncedSearchTerm]);
 
   return (
     <div className="space-y-4">
       <div className="space-y-1">
-        <div className="font-display font-medium text-foreground-1 text-3xl">
+        <div className="font-display text-3xl font-medium text-foreground-1">
           AVS
         </div>
-        <div className="font-display font-medium text-foreground-1 text-base">
-          Actively Validated Services
+        <div className="mb-6 flex w-full flex-col items-start justify-between gap-4 lg:flex-row lg:items-center">
+          <div className="font-display text-base font-medium text-foreground-1">
+            Actively Validated Services
+          </div>
+          <Input
+            classNames={{
+              inputWrapper:
+                'border border-outline data-[hover=true]:border-foreground-1',
+              input: 'placeholder:text-subtitle'
+            }}
+            value={state.searchTerm ?? ''}
+            onChange={handleSearch}
+            type="text"
+            placeholder="Search by AVS"
+            radius="sm"
+            className="lg:w-96"
+            variant="bordered"
+            endContent={
+              <span className="material-symbols-outlined text-subtitle">
```

Comment:
deprecating usage of `-subtitle`
```suggestion
              <span className="material-symbols-outlined text-foreground-2">
```